packages/agent/src/agent/http/index.ts

```diff
@@ -361,28 +361,41 @@ export class HttpAgent implements Agent {
   }
 
   private async _requestAndRetry(request: () => Promise<Response>, tries = 0): Promise<Response> {
-    const response = await request();
-    if (response.ok) {
-      return response;
-    }
+    try {
+      const response = await request();
+      if (response.ok) {
+        return response;
+      }
 
-    const responseText = await response.clone().text();
-    const errorMessage =
-      `Server returned an error:\n` +
-      `  Code: ${response.status} (${response.statusText})\n` +
-      `  Body: ${responseText}\n`;
+      const responseText = await response.clone().text();
+      const errorMessage =
+        `Server returned an error:\n` +
+        `  Code: ${response.status} (${response.statusText})\n` +
+        `  Body: ${responseText}\n`;
 
-    if (this._retryTimes > tries) {
-      console.warn(errorMessage + `  Retrying request.`);
-      return await this._requestAndRetry(request, tries + 1);
-    }
+      if (this._retryTimes > tries) {
+        console.warn(errorMessage + `  Retrying request.`);
+        return await this._requestAndRetry(request, tries + 1);
+      }
 
-    throw new AgentHTTPResponseError(errorMessage, {
-      ok: response.ok,
-      status: response.status,
-      statusText: response.statusText,
-      headers: httpHeadersTransform(response.headers),
-    });
+      throw new AgentHTTPResponseError(errorMessage, {
+        ok: response.ok,
+        status: response.status,
+        statusText: response.statusText,
+        headers: httpHeadersTransform(response.headers),
+      });
+    } catch (error) {
```

Comment:
It might make sense to put this try..catch around only the `await request()`, or otherwise consolidate it with the `response.ok` checking.  Consider what happens now after throw new AgentHTTPResponseError().